iidm/iidm-mergingview/src/main/java/com/powsybl/iidm/mergingview/LineAdderAdapter.java

```diff
@@ -0,0 +1,224 @@
+package com.powsybl.iidm.mergingview;
+
+import com.powsybl.commons.PowsyblException;
+import com.powsybl.iidm.network.*;
+
+import java.util.Objects;
+
+class LineAdderAdapter implements LineAdder {
+
+    private final MergingViewIndex index;
+
+    private String id;
+
+    private String name;
+
+    private boolean ensureIdUnicity;
+
+    private Integer node1;
+
+    private String bus1;
+
+    private String connectableBus1;
+
+    private String voltageLevelId1;
+
+    private Integer node2;
+
+    private String bus2;
+
+    private String connectableBus2;
+
+    private String voltageLevelId2;
+
+    private double r = Double.NaN;
+
+    private double x = Double.NaN;
+
+    private double g1 = Double.NaN;
+
+    private double b1 = Double.NaN;
+
+    private double g2 = Double.NaN;
+
+    private double b2 = Double.NaN;
+
+    LineAdderAdapter(final MergingViewIndex index) {
+        this.index = Objects.requireNonNull(index, "merging view index is null");
+    }
+
+    @Override
+    public Line add() {
+        Line newLine = null;
+        final Network n1 = checkAndGetNetwork1();
+        final Network n2 = checkAndGetNetwork2();
+        if (n1 == n2) {
+            newLine = index.getLine(addLine(n1));
+        } else {
+            // Creation of 2 dangling lines
+            throw MergingView.NOT_IMPLEMENTED_EXCEPTION;
+        }
+        return newLine;
+    }
+
+    private Line addLine(final Network network) {
+        final LineAdder adder = network.newLine()
+                .setId(id)
+                .setEnsureIdUnicity(ensureIdUnicity)
+                .setName(name)
+                .setR(r)
+                .setX(x)
+                .setG1(g1)
+                .setB1(b1)
+                .setG2(g2)
+                .setB2(b2)
+                .setVoltageLevel1(voltageLevelId1)
+                .setVoltageLevel2(voltageLevelId2);
+        if (network.getVoltageLevel(voltageLevelId1).getTopologyKind() == TopologyKind.BUS_BREAKER) {
+            adder.setBus1(bus1)
+                 .setConnectableBus1(connectableBus1);
+        } else {
+            adder.setNode1(node1);
+        }
+        if (network.getVoltageLevel(voltageLevelId2).getTopologyKind() == TopologyKind.BUS_BREAKER) {
+            adder.setBus2(bus2)
+                 .setConnectableBus2(connectableBus2);
+        } else {
+            adder.setNode2(node2);
+        }
+        return adder.add();
+    }
+
+    private Network checkAndGetNetwork1() {
+        if (voltageLevelId1 == null) {
+            throw new PowsyblException("first voltage level is not set");
+        }
+        Network network = index.getNetworkStream()
+                .filter(n -> n.getVoltageLevel(voltageLevelId1) != null)
+                .findFirst()
+                .orElse(null);
+        if (network == null) {
+            throw new PowsyblException("first voltage level '" + voltageLevelId1 + "' not found");
+        }
+        return network;
```

Comment:
To be discussed: if we have performance issue, we should consider to add a private-package method to get the underlying network instead of iterating over the networks.